src/common/lib/transport/connectionmanager.ts

```diff
@@ -21,6 +21,8 @@ import HttpStatusCodes from 'common/constants/HttpStatusCodes';
 type Realtime = any;
 type ClientOptions = any;
 
+let globalObject = typeof global !== 'undefined' ? global : typeof window !== 'undefined' ? window : self;
```

Comment:
Is there a reason we couldn't switch all of the use of `global` to [`globalThis`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/globalThis) instead?

Reply:
This is a good idea, it's a bit of effort to check how this affects platforms support. I don't know, for example, if `globalThis` is widely supported by webworkers since in webworkers it's more common to use `self`. I've opened #1401 to do this separately.